### PR TITLE
#10789 consolidating ctrl+enter settings into just one

### PR DESCRIPTION
**Pull request by carmo-evan**
#### Summary

Replace existing two settings related to CTRL+ENTER hotkey with one setting and three options within the expanded view of the setting.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/10789

### CODE REVIEW

**Comment by hanzei** [comment]
Hey @carmo-evan,

Thanks for the PR! In order to set up a test server, the build has to pass. It's currently failing because you have to add the new i18n strings to `en.json`. Let me know, if you have questions about this.

**Comment by carmo-evan** [comment]
> Hey @carmo-evan,
> 
> Thanks for the PR! In order to set up a test server, the build has to pass. It's currently failing because you have to add the new i18n strings to `en.json`. Let me know, if you have questions about this.

Thanks! Just added the string.

**Comment by carmo-evan** [comment]
@hanzei it still seems to be failing. I understand I probably have to remove these two unused strings: https://github.com/mattermost/mattermost-webapp/blob/9ad604ca8e0115cfaa0e827c092717056616e6a8/i18n/en.json#L2882-L2883
I tried it locally though, replicating [these steps](https://circleci.com/gh/mattermost/mattermost-webapp/2141?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link) but I still get a failure, if I'm reading it correctly:

image

Please advise.

File: i18n/en.json [hunk 1]
```diff
@@ -2896,6 +2896,7 @@
   "user.settings.advance.markdown_preview": "Show markdown preview option in message input box",
   "user.settings.advance.off": "Off",
   "user.settings.advance.on": "On",
+  "user.settings.advance.onForCode":"On for code-formatted messages (starting with ```)",
```
Comment (hanzei) [inline i18n/en.json]:
```suggestion
  "user.settings.advance.onForCode": "On for code-formatted messages (starting with ```)",
```

**Comment by hanzei** [comment]
> I understand I probably have to remove these two unused strings: 

That's correct.

See my other comment on how to fix the second failure.

**Comment by carmo-evan** [comment]
> > I understand I probably have to remove these two unused strings:
> 
> That's correct.
> 
> See my other comment on how to fix the second failure.
> 
> Would you also please merge `master` into this?

@hanzei all set.

**Comment by hanzei** [comment]
@esethna This is ready for review

**Comment by esethna** [comment]
Thanks @carmo-evan, functionally the PR is working great,

A couple issues found with the UI:

1. The default choice of "Off" is not selected the first time you open the setting. Notice no options are selected in the image below:
 image

2. The help text and settings labels seems to be different than specified in the ticket, I think we can improve them. Propose:

- `On for all messages`
- `On only for code blocks starting with ``` `
- `Off`

Help text: `When enabled, CTRL + ENTER will send the message and ENTER inserts a new line.`

**Comment by carmo-evan** [comment]
@hanzei @esethna this should be all set for a second review. Not sure if a new test server will spin up or if the old one will eventually get updated.

**Comment by carmo-evan** [comment]
@esethna @hanzei I just realized this latest change will depend on [this PR](https://github.com/mattermost/mattermost-redux/pull/838) to mattermost-redux getting merged.

**Comment by carmo-evan** [comment]
> @esethna @hanzei I just realized this latest change will depend on [this PR](https://github.com/mattermost/mattermost-redux/pull/838) to mattermost-redux getting merged.

@hanzei I chatted with @enahum over at mattermost-redux and removed the dependency, bringing the method into the web-app repo. Should be good to go.

**Comment by carmo-evan** [comment]
@hanzei I think there's something wrong with the test server, maybe because it was created during Github's outage yesterday. It's returning `404 page not found`.

**Comment by hanzei** [comment]
I don't know why the server fails to started. Asked for help in https://community.mattermost.com/core/pl/kd4p7r45cidg3edzpk6nbr6sfh.

**Comment by cpanato** [comment]
looks like the tar gz package was not built correctly. rerunning the job to check
fixed

**Comment by carmo-evan** [comment]
Gotcha. I’ll refactor.

**Pull request closed without merging**